src/storage/src/hummock/cache.rs

```diff
@@ -0,0 +1,664 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ptr::null_mut;
+use std::sync::Arc;
+
+use spin::Mutex;
+
+const IN_CACHE: u8 = 1;
+const REVERSE_IN_CACHE: u8 = !IN_CACHE;
+
+pub struct LruHandle<K: PartialEq + Default, T> {
```

Comment:
Can we have some doc for the struct in this doc? Like the overall idea, code structure and the explanation for struct fields and method parameters.